Bottom: eb70b4eb565e1ea8258ed8f9d9b934a43b4b7f7d
Top:    2137fd4bd686cabc01c898987f8ee28780bf7142
Author: Robert Collins <robertc@robertcollins.net>
Date:   2013-04-02 07:59:41 +1300

DecorateTestCaseResult was missing some used attributes in its exclusion of
forwarded attributes.


---

diff --git a/testtools/testcase.py b/testtools/testcase.py
index 1121ad2..628ddf2 100644
--- a/testtools/testcase.py
+++ b/testtools/testcase.py
@@ -906,8 +906,8 @@ class DecorateTestCaseResult(object):
         delattr(self.decorated, name)
 
     def __setattr__(self, name, value):
-        if name == 'decorated':
-            self.__dict__['decorated'] = value
+        if name in ('decorated', 'callout', 'before_run', 'after_run'):
+            self.__dict__[name] = value
             return
         setattr(self.decorated, name, value)
